fmralign/metrics.py

```diff
@@ -0,0 +1,166 @@
+    reconstruction error (Bazeille et al., 2019)
+
+    Parameters
+    ----------
+    ground_truth: 3D or 4D Niimg
+        Reference image
+    prediction : 3D or 4D Niimg
+        Same shape as `ground_truth`
+    masker: instance of NiftiMasker or MultiNiftiMasker
+        Masker to be used on ground_truth and prediction. For more information see:
+        http://nilearn.github.io/manipulating_images/masker_objects.html
+    loss : str in ['R2', 'corr', 'n_reconstruction_err']
+        The loss function used in scoring. Default is normalized
+        reconstruction error.
+        'R2' :
+            The R2 distance between source and target arrays.
+            Best possible score is 1.0 and it can be negative (because the
+            model can be arbitrarily worse).
+        'corr' :
+            The correlation between source and target arrays.
+        'n_reconstruction_err' :
+            The normalized reconstruction error. A perfect prediction
+            yields a value of 1.0
+    multioutput: str in ['raw_values', 'uniform_average']
+        Defines aggregating of multiple output scores. Default is raw values.
+        'raw_values' :
+            Returns a full set of scores in case of multioutput input.
+        'uniform_average' :
+            Scores of all outputs are averaged with uniform weight.
+
+    Returns
+    -------
+    score : float or ndarray of floats
+        The score or ndarray of scores if ‘multioutput’ is ‘raw_values’.
+        The worst possible score is arbitrarily set to -1 for all metrics.
+    """
+    X_gt = masker.transform(ground_truth)
+    X_pred = masker.transform(prediction)
+
+    if loss is "R2":
+        score = r2_score(X_gt, X_pred, multioutput=multioutput)
+    elif loss is "n_reconstruction_err":
+        score = normalized_reconstruction_error(
+            X_gt, X_pred, multioutput=multioutput)
+    elif loss is "corr":
+        score = np.array([pearsonr(X_gt[:, vox], X_pred[:, vox])[0]  # pearsonr returns both rho and p
+                          for vox in range(X_pred.shape[1])])
+        if multioutput == "uniform_average":
+            score = np.mean(score)
+    else:
+        raise NameError(
+            "Unknown loss. Recognized values are 'R2', 'corr', or 'reconstruction_err'")
+    # if the calculated score is less than -1, return -1
+    return np.maximum(score, -1)
+
+
+def normalized_reconstruction_error(y_true, y_pred, sample_weights=None,
+                                    multioutput='raw_values'):
+    """
+    Calculates the normalized reconstruction error
+    as defined by Bazeille and colleagues (2019).
+    
+    A perfect prediction yields a value of 1.
+    
+    Parameters
+    ----------
+    y_true : arr
+        The ground truth array.
+    y_pred : arr
+        The predicted array.
+    sample_weights : arr
+        Weights to assign to each sample.
+    multioutput: str in ['raw_values', 'uniform_average',
+                         'variance_weighted']
```

Comment:
```suggestion
]
```

Reply:
I committed this outside of the UI so resolving ! :smile_cat: